File: Arrays _101/1299. Replace Elements with Greatest Element on Right Side/replace_element.cpp
class Solution
{
public:
    vector<int> replaceElements(vector<int> &arr)
    {
        int max_val = arr[arr.size() - 1];
        arr[arr.size() - 1] = -1;
        int temp;

        for (int i = arr.size() - 2; i >= 0; i--)
        {
            temp = arr[i];
            arr[i] = max_val;
            max_val = max_val > temp ? max_val : temp;
        }
        return arr;
    }
};

File: Arrays _101/1299. Replace Elements with Greatest Element on Right Side/replace_element.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        max_val = arr[len(arr) - 1]
        arr[len(arr) - 1] = -1

        for i in range(len(arr) - 2, -1, -1):
            temp = arr[i]
            arr[i] = max_val
            max_val = max(max_val, temp)

        return arr